src/Adapter/Tornado/EventLoop.php

```diff
@@ -162,11 +195,24 @@ public function promiseForeach($traversable, callable $function): Promise
      */
     public function promiseRace(Promise ...$promises): Promise
     {
+        $cancellation = null;
+        $promisesCancellation = null;
+        $wrappedPromise = [];
         if (empty($promises)) {
             return $this->promiseFulfilled(null);
         }
 
-        $globalPromise = Internal\PendingPromise::createUnhandled($this->unhandledFailingPromises);
+        $promisesCancellation = function () use (&$wrappedPromise) {
+            foreach ($wrappedPromise as $index => $promise) {
+                $promise->cancel();
+            }
+        };
+
+        $cancellation = function () use (&$promisesCancellation) {
+            ($promisesCancellation)();
+        };
```

Comment:
When you call `$cancellation`, it calls `$promisesCancellation`… what about calling `$promisesCancellation` directly? 😄